navigation/screens/login.tsx

```diff
@@ -1,25 +1,63 @@
-import React, { useState } from 'react';
-import { StyleSheet,Image, SafeAreaView, TextInput, Button, Alert, TouchableOpacity, Text, View } from 'react-native';
-import { useNavigation } from '@react-navigation/native';
+import React, { useState } from "react";
+import {
+  StyleSheet,
+  Image,
+  SafeAreaView,
+  TextInput,
+  Button,
+  Alert,
+  TouchableOpacity,
+  Text,
+  View,
+} from "react-native";
+import { useNavigation } from "@react-navigation/native";
+import {
+  loginUser,
+  registerUser,
+  forgotPassword,
+} from "@/controllers/userController";
 
 function TabLoginScreen() {
   // State management for text inputs
+  const [firstname, setFirstname] = useState("");
+  const [lastname, setLastname] = useState("");
+  const [email, setEmail] = useState("");
   const [username, setUsername] = useState("");
   const [password, setPassword] = useState("");
+  const [profile, setProfile] = useState(Image);
```

Comment:
this is a correct use of state because these values change over the course of the lifetime of the component, in this case as the user types to fill out the form